src/Event/EventManager.php

```diff
@@ -330,7 +330,11 @@ protected function _callListener(callable $listener, EventInterface $event): voi
         $result = $listener($event, ...array_values($event->getData()));
 
         if ($result !== null) {
-            $class = get_class($event->getSubject());
+            try {
```

Comment:
Instead of using a try catch skip the `get_class()` call if `$event->getSubject()` returns null.

Reply:
its not about the `get_class` throwing, its about `getSubject()` throwing because no subject is present.
See https://github.com/cakephp/cakephp/blob/5.x/src/Event/Event.php#L113

Reply:
Oh, it's a bit strange that the method throws and exception instead of returning null when the subject is optional. Perhaps something to change in 6.0

Reply:
I'd also be in favor of just returning null.
This seems to be have done this way due to static analyzer problems in the past.
https://github.com/cakephp/cakephp/pull/13568